src/debug.ts

```diff
@@ -0,0 +1,158 @@
+import stringify from 'json-stringify-safe'
+import { Server } from 'node:http'
+import ws from 'ws'
+const { bootWebServers } = require('../web');
+
+type SharedSenecaState = {
+  expressIsReady: boolean
+  expressServer: Server | null
+  wsServer: ws.Server<ws.WebSocket> | null
+}
+
+function inward(seneca: any, spec: any, options: any) {
+  const { logToConsole, wspath } = options
+  const { data } = spec
+  data.debug_kind = 'in'
+
+  let data_in: any
+
+  if (!seneca.shared.expressIsReady) {
+    return
+  }
+
+  if (!options.prod) {
+    seneca.shared.wsServer!.emit(wspath, data)
+  }
+  data_in = stringify(data)
+
+  if (logToConsole) {
+    console.log(data_in)
+  }
+
+  data_in = JSON.parse(data_in)
+  if (options.prod) {
+    data_in.msg = seneca.util.clean(data_in.msg)
+    data_in.meta = {
+      id: data.meta.id,
+      start: data.meta.start,
+      end: data.meta.end,
+      pattern: data.meta.pattern,
+      plugin: data.meta.plugin,
+      instance: data.meta.instance
+    }
+
+    if (data.meta.parents && data.meta.parents.length > 1) {
+      data_in.meta.parents = null
+      data_in.meta.parent = data.meta.parents[0][1]
+    } else {
+      data_in.meta.parents = data.meta.parents
+      data_in.meta.custom = data.meta.custom
+    }
+  }
+
+  seneca.shared.wsServer!.clients.forEach((c: any) => {
+    c.send(JSON.stringify(data_in))
+  })
+}
+
+function outward(seneca: any, spec: any, options: any) {
+  const { logToConsole, wspath } = options
+  const { data } = spec
+  data.debug_kind = 'out'
+
+  let data_out: any
+
+  if (!seneca.shared.expressIsReady) {
+    return
+  }
+
+  if (!options.prod) {
```

Comment:
add a guard here: `if(!options.prod && seneca.shared.wsServer) {`
the `!` operation is dangerous and in this case, the warning feels valid